network/tcp_client_test.cc

```diff
@@ -0,0 +1,289 @@
+        }
+    }
+
+    void SetHandler(RequestHandler handler)
+    {
+        std::lock_guard<std::mutex> lock(m_handler_mutex);
+        m_handler = std::move(handler);
+    }
+
+private:
+    void RunLoop()
+    {
+        while (m_running && m_conn && m_conn->IsOpen())
+        {
+            auto msg = ReceiveSocketMessage(m_conn.get());
+            if (!msg.ok())
+            {
+                if (absl::IsDeadlineExceeded(msg.status()))
+                {
+                    continue;
+                }
+                break;
+            }
+
+            auto type = (*msg)->GetMessageType();
+            if (type == MessageType::HANDSHAKE_REQUEST)
+            {
+                HandshakeResponse resp;
+                resp.SetMajorVersion(1);
+                resp.SetMinorVersion(0);
+                if (!SendSocketMessage(m_conn.get(), resp).ok())
+                {
+                    break;
+                }
+            }
+            else if (type == MessageType::PING_MESSAGE)
+            {
+                PongMessage resp;
+                if (!SendSocketMessage(m_conn.get(), resp).ok())
+                {
+                    break;
+                }
+            }
+            else
+            {
+                std::lock_guard<std::mutex> lock(m_handler_mutex);
+                if (m_handler)
+                {
+                    m_handler(m_conn.get(), std::move(*msg));
+                }
+            }
+        }
+    }
+
+    std::shared_ptr<FakeSocketConnection> m_conn;
+    std::atomic<bool>                     m_running{ false };
+    std::thread                           m_thread;
+    RequestHandler                        m_handler;
+    std::mutex                            m_handler_mutex;
+};
+
+struct TestContext
+{
+    std::shared_ptr<FakeSocketConnection> server_socket;
+    FakeServer                            server;
+    std::unique_ptr<TcpClient>            client;
+
+    TestContext()
+    {
+        server_socket = std::make_shared<FakeSocketConnection>();
+
+        client = std::make_unique<TcpClient>([this]() {
+            auto connection = std::make_unique<FakeSocketConnection>();
+            connection->PairWith(server_socket.get());
+            return connection;
+        });
+
+        server.Start(server_socket);
+        EXPECT_TRUE(client->Connect("fake_host", 0).ok());
+    }
+
+    ~TestContext()
+    {
+        client->Disconnect();
+        server.Stop();
+    }
+};
+
+TEST(TcpClientFakeTest, GetCaptureFileSizeSuccess)
+{
+    TestContext      ctx;
+    constexpr size_t kExpectedSize = 12345;
+    MessageType      received_type = MessageType::UNKNOWN;
+
+    ctx.server.SetHandler([&](FakeSocketConnection* conn, std::unique_ptr<ISerializable> req) {
+        received_type = req->GetMessageType();
+
+        FileSizeResponse resp;
+        resp.SetFound(true);
+        resp.SetFileSizeStr(std::to_string(kExpectedSize));
+        (void)SendSocketMessage(conn, resp);
```

Comment:
nit: For absl::Status, the preferred way marking that the return type is unused this is with [IgnoreError()](https://abseil.io/docs/cpp/guides/status#ignoring-status-results)
```suggestion
        SendSocketMessage(conn, resp).IgnoreError();
```